src/map/pc.cpp

```diff
@@ -10437,6 +10479,28 @@ bool pc_setparam(map_session_data *sd,int64 type,int64 val_tmp)
 		sd->cook_mastery = val;
 		pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery);
 		return true;
+	case SP_GOLDPC_POINTS:
```

Comment:
solution for Polkye bad time manipulation bug report : https://github.com/rathena/rathena/pull/7410#issuecomment-1779162177
need to change all this case section to:
```suggestion
	case SP_GOLDPC_POINTS:
		val = cap_value(val, 0, battle_config.feature_goldpc_max_points);

		pc_setreg2(sd, GOLDPC_POINT_VAR, val);

		// If you do not check this, some funny things happen (circle logics, timer mismatches, etc...)
		if (!sd->state.connect_new) {
			// Make sure to always delete the timer
			const struct TimerData* td{};
			if (sd->goldpc_tid != INVALID_TIMER) {
				td = get_timer(sd->goldpc_tid);
				delete_timer(sd->goldpc_tid, pc_goldpc_update);
				sd->goldpc_tid = INVALID_TIMER;
			}
			// If the system is enabled and the player can still earn some points restart the timer
			if (battle_config.feature_goldpc_active && val < battle_config.feature_goldpc_max_points && !sd->state.autotrade) {
				// Capture the current time
				t_tick current_time = (td != NULL ? td->tick : 0) - gettick();
				current_time += (current_time % 1000);

				// Ensure remaining time is non-negative
				t_tick remaining_seconds = battle_config.feature_goldpc_time - current_time / 1000;
				remaining_seconds = (t_tick)max(static_cast<int>(remaining_seconds), 0);

				// Restart the timer with the remaining time
				sd->goldpc_tid = add_timer(gettick() + ((battle_config.feature_goldpc_time - remaining_seconds) * 1000 > 0 ? (battle_config.feature_goldpc_time - remaining_seconds) * 1000 : battle_config.feature_goldpc_time * 1000), pc_goldpc_update, sd->bl.id, (intptr_t)nullptr);
			}
			// Update the client
			clif_goldpc_info(*sd);
		}
		return true;
```